homepage/views.py

```diff
@@ -139,13 +140,49 @@ def index(request):
 
 
 def download(request, os=None):
-    os_response = {'os': None}
     if os is None:
-        os_response['os'] = detect_os(request.META.get('HTTP_USER_AGENT', ''))
+        os = detect_os(request.META.get('HTTP_USER_AGENT', ''))
+
+    return render(request, 'download.html', dict(
+        os=os,
+        tabs=[
+            dict(icon="windows", title="Windows", os="windows"),
+            dict(icon="apple", title="macOS", os="mac-os-x"),
+            dict(icon="linux", title="Linux / Source", os="linux"),
+        ],
+        recommended=recommended_download(os)
+    ))
+
+
+VERSION_RE = re.compile(r"Orange3-([\d\.]+)\.")
+
+
+def recommended_download(os):
+    downloads = download_choices()
+
+    def get_version(filename):
+        try:
+            return VERSION_RE.findall(filename)[0]
+        except IndexError:
+            return "unknown"
+
+    filename = title = ""
+    if os == "windows":
+        title = "Windows Installer"
+        filename = downloads["orange3-win32-installer"]
+    elif os == "mac-os-x":
+        title = "macOS bundle"
+        filename = downloads["bundle-orange3"]
+
+    if filename:
+        return dict(
+            filename=filename,
+            url=reverse('download') + 'files/' + filename,
+            version=get_version(filename),
+            title=title
+        )
     else:
-        os_response['os'] = os
-    os_response['generic_download_page'] = request.path.endswith('download/')
```

Comment:
There was a reason for this. This is the download page: http://orange.biolab.si/download/ For me, it is the same as http://orange.biolab.si/download/linux/, which is great, but this little hack kept the title neutral for Google. Could you make it above
```py
if os is None:
   # 302 redirect to detected os URL
```